llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -23,6 +23,15 @@
 using namespace llvm;
 using namespace LegalityPredicates;
 
+static LegalityPredicate typeIsLegalScalarFP(unsigned TypeIdx,
```

Comment:
When we add support for Zfh or Zfhmin in GISel, will you modify this function to check whether we `hasZfh || hasZfhmin`? What will happen if we only want to legalize an opcode when hasZfh but not when `!hasZfh && hasZfhmin`? Given these questions, is `typeIsLegalScalarFP` the correct name for this function?

Reply:
This function will only get Zfh added to it. `typeIsScalarFPArith`? G_FTRUNC/G_FPEXT are the only ones touched by this patch that need Zfhmin.

Reply:
I prefer typeIsScalarFPArith to typeIsLegalScalarFP since it doesn't have the word `legal` in it. I think the word legal is unnecessary since it will go into a `legalIf` which makes it redundant. I like ScalarFPArith better than ScalarFP since it will suggest that not all ScalarFP instructions use it.